internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -338,13 +293,21 @@ func (c *CodePipeline) getStage(s *cp.StageDeclaration) (*Stage, error) {
 		switch category {
 
 		case "Source":
-			// Currently, our only source provider is GitHub: https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html#structure-configuration-examples
-			details = fmt.Sprintf("Repository: %s/%s", aws.StringValue(config["Owner"]), aws.StringValue(config["Repo"]))
+			// https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html#structure-configuration-examples
+			switch provider {
+			case "GitHub":
+				details = fmt.Sprintf("Repository: %s/%s", aws.StringValue(config["Owner"]), aws.StringValue(config["Repo"]))
+			case "CodeCommit":
+				details = fmt.Sprintf("Repository: %s", aws.StringValue(config["RepositoryName"]))
+			case "CodeStarSourceConnection":
+				details = fmt.Sprintf("Repository: %s", aws.StringValue(config["FullRepositoryId"]))
+			default:
```

Comment:
can we just delete default if it's not used? (same for below)

Reply:
yup, done. for some reason I thought 'default' had to be there, but clearly not!

Reply:
Could the default case here be useful for error handling? Is there any chance of the input NOT being one of those values?

Reply:
No, we shouldn't be able to reach this point if it's not one of those.